examples/userapi/05_conditional_dimension.ipynb

```diff
@@ -0,0 +1,882 @@
+      "          }\n",
+      "        }\n",
+      "\n",
+      "    \n"
+     ]
+    }
+   ],
+   "source": [
+    "from devito import ConditionalDimension\n",
+    "print(ConditionalDimension.__doc__)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Use case A (honour a certain condition)\n",
+    "\n",
+    "\n",
+    "## Defining a condition\n",
+    "Now it's time to show a more descriptive example. Let's define a conditional that updates the values of the `Function` that are larger than zero, f(x,y) > 0. We name our `ConditionalDimension` `ci`.\n",
+    "In this example we want to update again by one all the elements in `f` that are larger than zero, $f(x,y) > 0$\n",
+    "Before updating the elements we initialize our data again to the eye function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Operator `Kernel` run in 0.01 s\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Data([[2., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n",
+       "      [1., 2., 1., 1., 1., 1., 1., 1., 1., 1.],\n",
+       "      [1., 1., 2., 1., 1., 1., 1., 1., 1., 1.],\n",
+       "      [1., 1., 1., 2., 1., 1., 1., 1., 1., 1.],\n",
+       "      [1., 1., 1., 1., 2., 1., 1., 1., 1., 1.],\n",
+       "      [1., 1., 1., 1., 1., 2., 1., 1., 1., 1.],\n",
+       "      [1., 1., 1., 1., 1., 1., 2., 1., 1., 1.],\n",
+       "      [1., 1., 1., 1., 1., 1., 1., 2., 1., 1.],\n",
+       "      [1., 1., 1., 1., 1., 1., 1., 1., 2., 1.],\n",
+       "      [1., 1., 1., 1., 1., 1., 1., 1., 1., 2.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "\n",
+    "from devito.types import Gt\n",
+    "\n",
+    "f.data[:, :] = np.eye(10)\n",
+    "\n",
+    "# Define the condition f(x, y) > 0\n",
+    "condition = Gt(f, 0)\n",
+    "\n",
+    "# Construct a ConditionalDimension\n",
+    "ci = ConditionalDimension(name='ci', parent=y, condition=condition)\n",
+    "\n",
+    "op = Operator(Eq(f, f + 1))\n",
+    "op.apply()\n",
+    "f.data\n",
+    "# print(op.ccode) # Uncomment to view code\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If we look at the generated code, we will see that it is identical to that from cell 3 -- no conditionals are generated. Why? We can notice that there are no conditionals in the generated code and as expected the function values are not updated. But why? We notice that `ci` is not used anywhere after construction. Neither `f` nor `op` know about `ci` yet. We then need to pass `ci` explicitly as an \"implicit `Dimension`\" to the equation that we want to be conditionally executed as shown in the next cell."
```

Comment:
Drop entirely: 

```
Why? We can notice that there are no conditionals in the generated code and as expected the function values are not updated. But 
```

(yes, including the 'But' after the full stop)

this sentence is redundant

Reply:
I suggesto to replace

```
why? We notice that `ci` is not used anywhere after construction
```

with

```
This is simply because we construct `ci` but we're not using it.
```

imho it sounds smoother

Reply:
Fixed